src/rsz/src/Rebuffer.cc

```diff
@@ -60,6 +60,423 @@ using sta::INF;
 using sta::NetConnectedPinIterator;
 using sta::PinSeq;
 
+// Template magic to make it easier to write algorithms descending
+// over the buffer tree in the form of lambdas; it allows recursive
+// lambda calling and it keeps track of the level number which is important
+// for good debug prints
+//
+// https://stackoverflow.com/questions/2067988/how-to-make-a-recursive-lambda
+template <class F>
+struct visitor
+{
+  F f;
+  int level = 0;
+  template <class... Args>
+  decltype(auto) operator()(Args&&... args)
+  {
+    level++;
+    decltype(auto) ret = f(*this, level, std::forward<Args>(args)...);
+    level--;
+    return ret;
+  }
+
+  // delete the copy constructor
+  visitor(const visitor&) = delete;
+  visitor& operator=(const visitor&) = delete;
+};
+template <typename F, class... Args>
+decltype(auto) visitTree(F&& f, Args&&... args)
+{
+  visitor<std::decay_t<F>> v{std::forward<F>(f)};
+  return v(std::forward<Args>(args)...);
+}
+
+void pruneCapVsSlackOptions(StaState* sta, BufferedNetSeq& Z)
+{
+  // Prune the options if there exists another option with
+  // larger required and smaller capacitance.
+  // This is fanout*log(fanout) if options are
+  // presorted to hit better options sooner.
+  std::unordered_map<BufferedNet*, Slack> slacks;
+
+  for (const BufferedNetPtr& p : Z) {
+    slacks[p.get()] = p->slack(sta);
+  }
+
+  sort(Z.begin(),
+       Z.end(),
+       [&slacks](const BufferedNetPtr& option1, const BufferedNetPtr& option2) {
+         const Slack slack1 = slacks[option1.get()];
+         const Slack slack2 = slacks[option2.get()];
+
+         if (slack1 != slack2) {
+           return slack1 > slack2;
+         }
+
+         return option1->cap() < option2->cap();
+       });
+
+  if (Z.empty()) {
+    return;
+  }
+
+  float Lsmall = Z[0]->cap();
```

Comment:
What does the L in Lsmall mean?  Likewise Lp below.  The variable naming convention is snake case (https://google.github.io/styleguide/cppguide.html#Variable_Names).

Reply:
Beats me, this is pre-existing code I copied around, I will try to rename